Use react-query for file listing

ReeceHoffmann opened the pull request:
Changes: 
 - Switches file manager views to use react-query
 - Switches Sample and subtraction creation view file listing to react-query
 - Adds websocket functionality for resources fetched via react-query
 - Add standardised factory functions for generating query keys
 - Fixed automatic test data generation to permit falsy custom values
 - Remove redux code relevant to listing files

ReeceHoffmann committed:
fix: prevent crash on file upload

 - Switches file manager views to use react-query
 -  Sample and subtraction creation view file listing moved to react-query
 - adds websocket functionality for resources fetched via react-query
- add standardised factory functions for generating query keys
- fixed automatic test generation data to permit falsy custom values
- remove redux code relevant to listing files

ReeceHoffmann committed:
chore: merge main into vir-1811

ReeceHoffmann committed:
chore: remove git diff markup

ReeceHoffmann committed:
test: remove obsolete snapshots

igboyes commented on src/js/app/websocket.ts:
```diff
@@ -13,6 +12,24 @@ import { wsInsertTask, wsUpdateTask } from "../tasks/actions";
+const reactQueryHandler = (queryClient: QueryClient) => (iface, operation) => {
+    const keyFactory = keyFactories[iface];
+    if (keyFactory === undefined) return;
+    queryClient.invalidateQueries(keyFactory.all);
```
```suggestion
    const keyFactory = keyFactories[iface];
    if (keyFactory) {
        return queryClient.invalidateQueries(keyFactory.all);
    }

```

igboyes commented on src/js/administration/querys.ts:
```diff
@@ -2,12 +2,24 @@ import { useMutation, useQuery, useQueryClient } from "react-query";
+    lists: () => [...userKeys.all, "list"] as const,
+    list: (filters: Array<string | number | boolean>) => [...userKeys.lists(), ...filters] as const,
+    details: () => [...userKeys.all, "details"] as const,
+    detail: (user_id: string) => [...userKeys.details(), user_id] as const,
```
It is difficult to understand what is going on here at a glance.

I'm not sure of the benefit of these `somethingKeys` objects. It seems that each property is only read in a single hook (eg. `useGetUsers`) and that the hook is the best place to define the keys once.

Thoughts?

ReeceHoffmann committed:
chore: address review

 - Simplifies react query key factory functions
 - Converts some arrow functions to function

ReeceHoffmann committed:
chore: increase usage of function notation

ReeceHoffmann committed:
Merge branch 'main' into reecehoffmann/vir-1811-upload-view-does-not-update-when-an-upload-completes

ghost commented:
:tada: This PR is included in version 3.12.2 :tada:

The release is available on [GitHub release](https://github.com/virtool/virtool-ui/releases/tag/3.12.2)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
